### PR TITLE
Replace HTTP calls with gRPC via micro proxy

### PR DESCRIPTION
**Pull request by ben-toogood**
Do not merge until [this](https://github.com/micro/go-micro/pull/1230/files) PR has been merged. Currently go.micro.auth is expecting a token and we're providing a JWT.

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -55,31 +55,34 @@ func run(ctx *cli.Context, srvOpts ...micro.Option) {
...
-		os.Exit(1)
-	}
-	buff := bytes.NewBuffer(jsonStr)
-
-	// Execute the HTTP request
-	resp, err := http.Post("https://api.micro.mu/auth/validate", "application/json", buff)
-	if err != nil || resp.StatusCode != 200 {
+		fmt.Println(err)
 		fmt.Println("Invalid token. Visit https://micro.mu/platform to get a token.")
 		os.Exit(1)
```
Comment (asim) [inline auth/auth.go]: Remove line "Invalid token. Visit https://micro.mu/platform to get a token."

Assume micro is self contained and if someone is using the open source and authenticating they should just see a go.micro.auth error.

File: runtime/service.go [hunk 2]
```diff
@@ -50,6 +50,16 @@ func defaultEnv() []string {
 	return env
 }
 
+func runtimeFromContext(ctx *cli.Context) runtime.Runtime {
+	if ctx.Bool("platform") {
+		os.Setenv("MICRO_PROXY", "go.micro.network")
```
Comment (asim) [inline runtime/service.go]: MICRO_PROXY=service

**Comment by ben-toogood** [comment]
@asim can you please review changes

File: auth/auth.go [hunk 3]
```diff
@@ -55,31 +55,33 @@ func run(ctx *cli.Context, srvOpts ...micro.Option) {
...
+	}
+
+	return *cmd.DefaultCmd.Options().Auth
+}
+
 // login using a token
 func login(ctx *cli.Context) {
 	if ctx.Args().Len() != 1 {
-		fmt.Println("Usage: `micro login [token]`")
+		fmt.Println("Usage: `micro login [token]. Visit https://micro.mu/platform to get a token.`")
```
Comment (asim) [inline auth/auth.go]: I would not add this. While we have a platform flag we want no reference to our product platform. This is open source and should be self contained.

**Pull request merged**